packages/instantsearch.js/src/lib/__tests__/server.test.ts

```diff
@@ -239,4 +249,52 @@ describe('getInitialResults', () => {
       },
     });
   });
+
+  test('returns the current results with request params if specified', async () => {
```

Comment:
we need another test with two indices with the same indexName and indexId that aren't adjacent too I think. I wonder if the deduplication could cause issues

Reply:
if you walk through the indices, you should be able to "skip back" to a previous result if you store the "request ids" in order right?


pseudo code for how that could work in the hydration of index widgets phase:

```js
let currentIndex = 0
let seenIndexIds: Record<IndexId, number> = {};
walkIndex(({indexName, indexId}) => {
  currentIndex = seenIndexIds[indexId] ?? currentIndex + 1;
  seenIndexIds[indexId] = currentIndex
})
```

Reply:
I looked at it and it's actually working as expected: there are as many `requestParams` as there are `walkIndex()` iterations.

The reason there's a mismatch later on is actually unrelated, `initialResults` is an object that is keyed by index ids, so the same index ids in later iterations overwrite previous ones.

I don't know if we should be more restrictive about having multiple identical index ids, but at least I can it knowingly and comment on the context.

Reply:
Hmm that makes sense. I guess we can separately make sure initialResults contains the combined results / uiState, and not accidentally only those of one instance